### PR TITLE
feat: AU-2548: Added new AD-group <-> Drupal role mappings.

### PR DESCRIPTION
**Pull request by Alexander-Tallqvist-Siili**
# [AU-2548](https://helsinkisolutionoffice.atlassian.net/browse/AU-2548)

## What was done

This pull request implements a few new AD-group <-> Drupal role mappings according to the instructions in the ticket. It also implements AD-group <-> INDUSTRY mappings according to the AD-group names.

## Other fixes

Fixed and issue in `grants_metadata.module` where no users other than the ones with the `grants_admin` role could edit anything on a Webforms third-party settings page.

## How to install

Make sure your instance is up and running on correct branch.
* `git checkout feature/AU-2544-new-ad-group-drupal-role-mappings`
* `make fresh`
* `make drush-cr`

## How to test

- [ ] Check that the new AD-group <-> Drupal role mappings are set according to [this](https://helsinkisolutionoffice.atlassian.net/wiki/spaces/KAN/pages/8831107145/AD-ryhm+t).
- [ ] Check that the new AD-group <-> INDUSTRY mappings are set according to [this](https://helsinkisolutionoffice.atlassian.net/wiki/spaces/KAN/pages/8831107145/AD-ryhm+t).
- [ ] Check that the roles `grant_producer` and `grant_producer_industry` can access the correct settings on Webforms.
- [ ] Take this to the staging environment and let Tero / Pirjo test it.

[AU-2544]: https://helsinkisolutionoffice.atlassian.net/browse/AU-2544?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[AU-2548]: https://helsinkisolutionoffice.atlassian.net/browse/AU-2548?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: public/modules/custom/grants_industries/grants_industries.module [hunk 1]
```diff
@@ -227,6 +226,16 @@ function grants_industries_openid_connect_post_authorize(UserInterface $account,
       if (str_contains($ad_group_lower, 'ta_sote')) {
         $industryKey = 'SOTE';
       }
+      // New mappings.
+      if (str_contains($ad_group_lower, 'laaja_Kanslia') || str_contains($ad_group_lower, 'suppea_Kanslia')) {
```
Comment (oskarimysteerio) [inline public/modules/custom/grants_industries/grants_industries.module:230]: Nää verrattavat stringit pitäis varmaan kans olla lower casella että mätsää?

**Review by oskarimysteerio — CHANGES_REQUESTED** [review]
Muuten varmana ok, mut noi lowercase vertailut pitää tarkistaa ja phpcs huutelee jotakin

File: public/modules/custom/grants_industries/grants_industries.module (hunk 1 above)
Comment (Alexander-Tallqvist-Siili) [reply public/modules/custom/grants_industries/grants_industries.module]: Hyvä spotti! Korjasin.

**Comment by Alexander-Tallqvist-Siili** [comment]
Korjattu PHPCS virheet.

**Review by oskarimysteerio — APPROVED** [review]

**Comment by Alexander-Tallqvist-Siili** [comment]
Jätän tän vielä @jiisuominen katottavaks niin, että pysyy kärryillä näistäkin asioista kun me muut ollaan poissa.

**Pull request merged**